lib/openstack_query/query_output.py

```diff
@@ -78,22 +82,25 @@ def parse_select(self, *props: PropEnum, select_all=False) -> None:
         :param props: any number of Enums representing properties to show
         """
         if select_all:
-            self._props = set(self._prop_handler.all_props())
-        else:
-            for prop in props:
-                self._check_prop_valid(prop)
-                self._props.add(prop)
+            self._props = set(self._prop_enum_cls)
+            return
+
+        for prop in props:
+            if prop not in self._prop_enum_cls:
+                raise ParseQueryError(
+                    f"Error: Given property to select: {prop.name} is not supported by query"
+                )
+
+            self._props.add(prop)
```

Comment:
EAFP this to keep it simple:

```
try:
   for prop in props:
      self._props.add(prop)
except KeyError:
  raise ParseQueryError(
                    f"Error: Given property to select: {prop.name} is not supported by query"
                )
```

Reply:
adding it is not going to fail - we need to test that it exists in prop_enum_cls first - no way to EAFP